src/main.cpp

```diff
@@ -46,6 +50,24 @@ int main(int argc, char** argv) {
             std::cout << "[ WARNING ] Running Hyprland with superuser privileges might damage your system\n";
 
             ignoreSudo = true;
+        } else if (it->compare("--socket") == 0) {
+            if (std::next(it) == args.end()) {
+                help();
+
+                return 1;
+            }
+
+            socketName = *std::next(it);
+            it++;
+        } else if (it->compare("--wayland-fd") == 0) {
+            if (std::next(it) == args.end()) {
+                help();
+
+                return 1;
+            }
+
+            socketFd = std::atoi(std::next(it)->c_str());
```

Comment:
ah wait a sec, this can crash without a useful message. Wrap it in a try cath

Reply:
Good catch, I didn't think about error cases much when I wrote the original PoC and didn't get to cleaning this up yet. Though, I think a try catch is not sufficient here: `std::atoi()` returns 0 on invalid numbers and causes undefined behaviour if the number is out of range. I should have used `std::stoi()` which does throw an exception.

Reply:
Fixed. I also added a check that `socketFd` refers to a valid file descriptor, and isn't just a random number.